client/network/src/lib.rs

```diff
@@ -137,9 +163,46 @@
 //! substream open with is chosen, and the information is requested from it.
 //! - Gossiping. Used for example by grandpa.
 //!
-//! It is intended that in the future each of these components gets more isolated, so that they
-//! are free to open and close their own substreams, and so that syncing and light client requests
-//! are able to communicate with nodes outside of the range of the PSM.
+//! ## Request-response protocols
+//!
+//! A so-called request-response protocol is defined as follow:
+//!
+//! - When a substream is open, the opening side sends a message whose content is
```

Comment:
```suggestion
//! - When a substream is opened, the opening side sends a message whose content is
```

This ensures one does not confuse edge triggering with level triggering.

Edge triggered actions would happen once (right after the substream was **opened**).

Level triggered actions would happen constantly (as long as the substream is **open**).